### PR TITLE
Setup Wizard: Prepare first setup wizard UI prototype

### PR DESCRIPTION
**Pull request by vovakulikov**
Part of https://github.com/sourcegraph/sourcegraph/issues/47237

This PR connected different parts of setup wizard, 
- Hide local repositories step (if you're running non Sourcegraph app instance)
- Connect next button with number of connected code hosts (you're not able to go to the next step until you connect at least one code host)
- Add a few new supported code host in the main remote code hosts list
- Connects orgs and repositories picker to newly merged API 

A few important parts will be resolved in follow PRs
- Add access token async validation, (it turned out that access token query hasn't been merged yet, we discussed it with Garry and it seems that there is a way to get this query soon, if something will show up we will use a workaround for this)
- Edit UI pickers doesn't work properly, also this is because pickers API for the edit UI haven't been merged yet

## Test plan
- Run you instance and
- Enable setup wizard feature flat `experimentalFeatures: { setupWizard: true }`
- Go to the `/setup` URL 
- Try to connect one of available code host 
- If you're connecting Github code host it should be possible to use form UI instead of code host json editor option 

## App preview:

- [Web](https://sg-web-vk-prepare-first-setup-prototype.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.07 kb) | 0.01% (+1.91 kb) | 0.02% (+1.84 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/vk/prepare-first-setup-prototype/compare-report.html;tab=live_object) for a full comparison between the commits 906a8a0 and 6015496 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/setup-wizard/components/remote-repositories-step/components/code-hosts/github/GithubEntityPickers.tsx [hunk 1]
```diff
@@ -56,29 +85,63 @@ export const GithubOrganizationsPicker: FC<GithubOrganizationsPickerProps> = pro
...
 
-const DEMO_REPOS_SUGGESTIONS = [
-    'sourcegraph/sourcegraph',
-    'sourcegraph/about',
-    'personal/my-project',
-    'peraonal/university-labs',
-    'facebook/react',
-]
+const GET_GITHUB_REPOSITORIES = gql`
+    query GetGitHubRepositories($first: Int!, $token: String!, $query: String!, $excludeRepositories: [String!]!) {
```
Comment (st0nebraker) [inline client/web/src/setup-wizard/components/remote-repositories-step/components/code-hosts/github/GithubEntityPickers.tsx:95]: I see from the description this UI isn't hooked in yet. But I'm seeing this request fail on the browser with this error `"Cannot query field \"externalServiceRepositories\" on type \"Query\". Did you mean \"externalServiceNamespaces\", or \"externalServices\"?"`. Is this known?

File: client/web/src/setup-wizard/components/remote-repositories-step/helpers.ts [hunk 2]
```diff
@@ -30,8 +34,14 @@ export const getCodeHostName = (codeHostType: ExternalServiceKind | null): strin
...
         case ExternalServiceKind.BITBUCKETCLOUD:
             return 'BitBucket.org'
+        case ExternalServiceKind.BITBUCKETSERVER:
+            return 'BitBucket Server'
         case ExternalServiceKind.AWSCODECOMMIT:
             return 'AWS Code Commit'
+        case ExternalServiceKind.GITOLITE:
+            return 'Gitolite'
+        case ExternalServiceKind.GERRIT:
+            return 'Gerrit'
```
Comment (st0nebraker) [inline client/web/src/setup-wizard/components/remote-repositories-step/helpers.ts:44]: Is there a logo we should pair with `Gerrit`? Or the default git icon at least?
image

**Review by st0nebraker — APPROVED** [review]
Looks good to me. I see the organization network request going through, but no the repository request as noted below.

I'll save QA-ing the full GitHub org/repo picker for when that logic is complete. Same with other code hosts, in the favor of time, as I need to set up an account or find the testing repos for them.

File: client/web/src/setup-wizard/components/remote-repositories-step/components/code-hosts/github/GithubEntityPickers.tsx (hunk 1 above)
Comment (vovakulikov) [reply client/web/src/setup-wizard/components/remote-repositories-step/components/code-hosts/github/GithubEntityPickers.tsx:95]: Hmm, I think BE changes haven't been deployed on k8s yet, I think something happened with deploy there, it should works fine with s2 though

File: client/web/src/setup-wizard/components/remote-repositories-step/helpers.ts (hunk 2 above)
Comment (vovakulikov) [reply client/web/src/setup-wizard/components/remote-repositories-step/helpers.ts:44]: Yeah, it turned out that Gerrit has a custom icon that we drew at some point and this makes the whole helper about icons useless if we want to have something custom there, I will revisit these icons in a follow up polishing PR

File: client/web/src/setup-wizard/components/remote-repositories-step/components/code-hosts/github/GithubEntityPickers.tsx (hunk 1 above)
Comment (st0nebraker) [reply client/web/src/setup-wizard/components/remote-repositories-step/components/code-hosts/github/GithubEntityPickers.tsx:95]: Ah I see it on S2, thanks!

**Pull request merged**